### ISSUE TITLE
GRPO reward func

### ISSUE THREAD

**Issue by sdpkjc** [huggingface/trl#2644]
👋 I noticed that the current GRPO reward function can only calculate rewards based on the prompt and completion. This allows for replicating the Format rewards mentioned in the original DeepSeek paper but makes it impossible to implement Accuracy rewards since it doesn’t have access to the ground truth information from the dataset. Are there any plans for further improvements? Or did I miss something? 🤔

**Comment by qgallouedec** [comment]
Yes, that's right, that's what I'm implementating today 🙂

**Comment by sdpkjc** [comment]
Great to hear that you're working on this—looking forward to the results! A potential structure for a customizable reward function could be: `def reward_func(completions, **kwargs)`, where `kwargs` contains the relevant dataset keys required by the reward function. Alternatively, the entire row of dataset information could be passed to the function, like this: `def reward_func(completions, dataset_row)`. However, this approach would make the inputs for the reward function and the reward model different, which might introduce inconsistencies.

**Comment by qgallouedec** [comment]
Yes, I was considering the first option. It's good to read that I'm not the only one to have had the idea of implementing it this way, so it must make sense.

**Comment by ymroueh** [comment]
Hey @qgallouedec thank you for putting together this :) !! I have one question in line https://github.com/huggingface/trl/blob/main/trl/trainer/grpo_trainer.py#L386 are you considering adding the min of this and  its clipping after this step as done in the orginal GRPO paper ?

**Comment by qgallouedec** [comment]
> are you considering adding the min of this and its clipping after this step as done in the orginal GRPO paper ?

Discussed here https://github.com/huggingface/trl/issues/2608#issuecomment-2609844003

**Comment by ymroueh** [comment]
thank you !

**Comment by zhibo** [comment]
I have an additional question. My model has a logits output based on a special_token. Is there a way to incorporate this into the input parameters of the reward_func?
